### DISCUSSION TITLE
Simultaneous Document DB Updates

### DISCUSSION THREAD

**Discussion by DCMattyG** [fastapi/fastapi#4786] [Questions] [answered]
Hi all, I'm wondering how others have dealt with examples such as an API endpoint that updates (performs an upsert) on a Document in something like MongoDB or perhaps CosmosDB.

I feel like a logic flow of reading the document, updating the data element, then upserting the new document is flawed because if 2 (or more) users hit that endpoint at the same time, that could result in some challenging conflicts.

How are others handling this or similar situations?

**Comment by jgould22** [comment, marked as answer]
I have never use CosmosDB. 

For Mongo here are the rules it follows. 
https://www.mongodb.com/docs/realm/sync/learn/conflict-resolution/#rules-of-conflict-resolution

Databases are not magic and usually operate on a series of trade offs.

If you need functionality like this you would probably be better served by an ACID compliant RDMS (Like PostgreSQL)

Mongodb is almost always the wrong choice, but it is fast and easy to prototype with so people choose it even though they would be better served with another DB.
